clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -229,7 +229,7 @@ public long connectionDelay(Node node, long now) {
      */
     @Override
     public boolean connectionFailed(Node node) {
-        return connectionStates.connectionState(node.idString()).equals(ConnectionState.DISCONNECTED);
+        return connectionStates.isDisconnected(node.idString());
```

Comment:
I agree it's weird. An alternative in `ConsumerNetworkClient` is to keep a set of the nodes that are currently connected and just use that when checking failed connections. That would also eliminate the annoying copy in `UnsentRequests.nodes()`.